pint/utils.py

```diff
@@ -439,6 +440,74 @@ def is_number(s):
         pass
     return False
 
+def make_toas(startMJD, endMJD, ntoas, model, freq=1400, obs='GBT'):
+    '''make evenly spaced toas without errors'''
+    def get_freq_array(bfv, ntoas):
+        freq = np.zeros(ntoas)
+        num_freqs = len(bfv)
+        for ii, fv in enumerate(bfv):
+            freq[ii::num_freqs] = fv
+        return freq
+    
+    times = np.linspace(np.longdouble(startMJD)*u.d, np.longdouble(endMJD)*u.d, ntoas) * u.day
+    freq_array = get_freq_array(np.atleast_1d(freq)*u.MHz, len(times))
+    t1 = [pint.toa.TOA(t.value, obs = obs, freq=f, \
+    scale=pint.observatory.get_observatory(obs).timescale) for t, f in zip(times, freq_array)]
+    ts = pint.toa.TOAs(toalist=t1)
+    ts.compute_TDBs()
+    ts.compute_posvels()
+    ts.clock_corr_info.update({'include_bipm':False,'bipm_version':'BIPM2015','include_gps':False})
+    return ts
+
+def show_param_cov_matrix(matrix,params,name='Covaraince Matrix',switchRD=False):
+    '''function to print covariance matrices in a clean and easily readable way'''
```

Comment:
Much better to have it generate and return a string that the user can then print, or log, or display in a user interface. This can be done using a StringIO and print(..., file=f).

Reply:
TODO added